java-checks/src/main/java/org/sonar/java/checks/UndocumentedApiCheck.java

```diff
@@ -198,11 +211,26 @@ private String className() {
     return className;
   }
 
-  private WildcardPattern[] getPatterns() {
-    if (patterns == null) {
-      patterns = PatternUtils.createPatterns(forClasses);
+  private WildcardPattern[] getInclusionPatterns() {
+    if (inclusionPatterns == null) {
+      if (StringUtils.isEmpty(forClasses)) {
+        forClasses = "**";
+      }
+
+      inclusionPatterns = PatternUtils.createPatterns(forClasses);
+    }
+    return inclusionPatterns;
+  }
+
+  private WildcardPattern[] getExclusionPatterns() {
+    if (exclusionPatterns == null) {
+      if (StringUtils.isEmpty(exclusion)) {
+        return new WildcardPattern[0];
```

Comment:
You should also lazily initialize the exclusionPatterns with an empty array when exclusion is not provided, as it will slightly improve performance when calling the rule on multiple files during analysis.